cardinal/types/engine/query.go

```diff
@@ -1,25 +1,3 @@
 package engine
 
-type Query interface {
-	// Name returns the name of the query.
-	Name() string
-	// Group returns the group of the query.
-	Group() string
-	// HandleQuery handles queries with concrete types, rather than encoded bytes.
-	HandleQuery(Context, any) (any, error)
-	// HandleQueryRaw is given a reference to the engine, json encoded bytes that represent a query request
-	// and is expected to return a json encoded response struct.
-	HandleQueryRaw(Context, []byte) ([]byte, error)
-	// DecodeEVMRequest decodes bytes originating from the evm into the request type, which will be ABI encoded.
-	DecodeEVMRequest([]byte) (any, error)
-	// EncodeEVMReply encodes the reply as an abi encoded struct.
-	EncodeEVMReply(any) ([]byte, error)
-	// DecodeEVMReply decodes EVM reply bytes, into the concrete go reply type.
-	DecodeEVMReply([]byte) (any, error)
-	// EncodeAsABI encodes a go struct in abi format. This is mostly used for testing.
-	EncodeAsABI(any) ([]byte, error)
-	// IsEVMCompatible reports if the query is able to be sent from the EVM.
-	IsEVMCompatible() bool
-	// GetRequestFieldInformation returns a map of the fields of the query's request type and their types.
-	GetRequestFieldInformation() map[string]any
-}
+type QueryHandler = func(name string, group string, bz []byte) ([]byte, error)
```

Comment:
Why cant this be in the cardinal package?

Reply:
Actually looks like that type isn't even used anywhere. Will remove.